### ISSUE TITLE
[test failed] [1.10 upgrade] Cadvisor should be healthy on every node

### ISSUE THREAD

**Issue by krzyzacy** [kubernetes/kubernetes#60768]
the tests are failing in http://k8s-testgrid.appspot.com/sig-release-master-upgrade#gce-1.9-master-upgrade-master

/sig sig-instrumentation
/priority failing-test
/priority critical-urgent
/kind bug
/status approved-for-milestone

cc @jdumars @jberkus
/assign @piosz @fabxc
also cc @dashpole for cadvisor I guess

**Comment by krzyzacy** [comment]
xref https://github.com/kubernetes/kubernetes/issues/60764

**Comment by tpepper** [comment]
@kubernetes/sig-release-admins looks like this needs /milestone'd by somebody with priv's

**Comment by krzyzacy** [comment]
whoops, thanks for reminding me

**Comment by brancz** [comment]
I dug into this a bit and I think I figured out what's happening.

The test logs of each of the 6 retries of the cAdvisor "ping" on each node says:

```
I0306 16:16:11.292] Mar  6 16:16:11.292: INFO: failed to retrieve kubelet stats -
I0306 16:16:11.293]  [the server could not find the requested resource the server could not find the requested resource the server could not find the requested resource the server could not find the requested resource]
I0306 16:16:21.292] [1mSTEP[0m: Querying stats from node bootstrap-e2e-master using url api/v1/proxy/nodes/bootstrap-e2e-master/stats/
I0306 16:16:21.295] [1mSTEP[0m: Querying stats from node bootstrap-e2e-minion-group-dspf using url api/v1/proxy/nodes/bootstrap-e2e-minion-group-dspf/stats/
I0306 16:16:21.296] [1mSTEP[0m: Querying stats from node bootstrap-e2e-minion-group-p1c4 using url api/v1/proxy/nodes/bootstrap-e2e-minion-group-p1c4/stats/
I0306 16:16:21.298] [1mSTEP[0m: Querying stats from node bootstrap-e2e-minion-group-vw6b using url api/v1/proxy/nodes/bootstrap-e2e-minion-group-vw6b/stats/
I0306 16:16:21.299] Mar  6 16:16:21.298: INFO: failed to retrieve kubelet stats -
I0306 16:16:21.299]  [the server could not find the requested resource the server could not find the requested resource the server could not find the requested resource the server could not find the requested resource]
```

At first this seemed curious, but I realized that this is actually the correct behavior for a 1.10 cluster, as the `/api/v1/proxy` API [has been removed for 1.10](https://github.com/kubernetes/kubernetes/pull/59884). As this is an upgrade test, however, the tests are executed (from how I understand) from latest 1.9.x release, where the path was indeed still using the old API.

I talked to @sttts, as I wasn't sure how to proceed, he told me, that we need to get the patch to use the "new"/non-deprecated API into a 1.9.x release, cut the release, and then the upgrade tests should pass. He also mentioned that you @mbohlool are the release lead for 1.9, I'm just tagging you so you are aware of this. I will open a PR against the 1.9 release branch to adapt the test to use the "new"/non-deprecated API.

**Comment by brancz** [comment]
Opened #60921, I'm not sure if I followed the procedure correctly, please let me know if I did something wrong.

**Comment by DirectXMan12** [comment]
/status in-progress

helping out since @brancz is a SIG lead ;-)

**Comment by liggitt** [comment]
in progress, 1.9 PR open at https://github.com/kubernetes/kubernetes/pull/60921

**Comment by liggitt** [comment]
fixed by #60921
